### PR TITLE
[DOC] Use SurfaceImage object for the Searchlight surface example

### LINKED ISSUE #4654: update surface searchlight to run with the new SurfaceImage
see https://nilearn.github.io/stable/auto_examples/02_decoding/plot_haxby_searchlight_surface.html#sphx-glr-auto-examples-02-decoding-plot-haxby-searchlight-surface-py

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4654

Changes proposed in this pull request:

- adapt searchlight on surface example to use the SurfaceImage object

### CODE REVIEW

File: examples/02_decoding/plot_haxby_searchlight.py
Comment (Remi-Gau) [inline examples/02_decoding/plot_haxby_searchlight.py:1]: unrelated but slight reorganization of the volume based searchlight example to show the searchlight results before doing the F test

File: examples/02_decoding/plot_haxby_searchlight_surface.py
Comment (Remi-Gau) [inline examples/02_decoding/plot_haxby_searchlight_surface.py:81]: note that the previous example only ran the left hemisphere and now it runs both, so this run time of the example will be doubled

made it possible to only run it on one hemisphere

File: examples/02_decoding/plot_haxby_searchlight_surface.py
Comment (Remi-Gau) [inline examples/02_decoding/plot_haxby_searchlight_surface.py:94]: should be possible to speed up the example by making the "sphere" around each vertex bigger

could also be useful for users to know how to do this

@bthirion @man-shu any thoughts?

**Comment by Remi-Gau** [comment]
@bthirion 
I think that for the coming release this is enough as it shows how to use the surface image object to run surface analysis.

I don't think that for now we want to make the SearchLight estimator accept SurfaceImage objects:
https://github.com/nilearn/nilearn/blob/8c4e2e59fc25bedc46d196922751765cd881e3c6/nilearn/decoding/searchlight.py#L237

File: examples/02_decoding/plot_haxby_searchlight_surface.py
Comment (bthirion) [reply examples/02_decoding/plot_haxby_searchlight_surface.py:16]: Indeed.

File: examples/02_decoding/plot_haxby_searchlight_surface.py
Comment (bthirion) [reply examples/02_decoding/plot_haxby_searchlight_surface.py:81]: I would run only one hemispheres to save resources.

File: examples/02_decoding/plot_haxby_searchlight_surface.py
Comment (bthirion) [reply examples/02_decoding/plot_haxby_searchlight_surface.py:94]: Increasing the radius will make it slower...

**Review by bthirion — COMMENTED** [review]
This looks great, thx ! 
I would run it only one one hemisphere only to spare resources.

File: examples/02_decoding/plot_haxby_searchlight_surface.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_haxby_searchlight_surface.py]: fixed: using the default radius now

File: examples/02_decoding/plot_haxby_searchlight_surface.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_haxby_searchlight_surface.py]: done

File: examples/02_decoding/plot_haxby_searchlight_surface.py
Comment (man-shu) [inline examples/02_decoding/plot_haxby_searchlight_surface.py:16]: As discussed in dev meeting, we need to replace the Haxby dataset with properly coregistered dataset for this example.

(Could not find the issue where this was initially discussed hence re-posting it here to further ref in a new issue)

**Review by man-shu — APPROVED** [review]
LGTM!

File: examples/02_decoding/plot_haxby_searchlight_surface.py
Comment (Remi-Gau) [reply examples/02_decoding/plot_haxby_searchlight_surface.py:16]: - relates to https://github.com/nilearn/nilearn/issues/4561

**Comment by Remi-Gau** [comment]
will wait for CI to finish before merging

**Pull request merged**
